libs/auth/src/drivers/magento/login.service.ts

```diff
@@ -0,0 +1,48 @@
+import { Injectable, Inject } from '@angular/core';
+import { Observable, pipe } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { Apollo } from 'apollo-angular';
+
+import { DaffLoginServiceInterface } from '../interfaces/login-service.interface';
+import { DaffAuthQueryManager } from '../injection-tokens/auth-query-manager.token';
+import { DaffAuthQueryManagerInterface } from '../interfaces/auth-query-manager.interface';
+import { GenerateTokenResponse } from './models/outputs/generate-token-response';
+import { DaffLoginInfo } from '../../models/login-info';
+import { DaffAuthTransformer } from '../injection-tokens/auth-transformer.token';
+import { DaffAuthTransformerInterface } from '../interfaces/auth-transformer.interface';
+import { DaffAuthToken } from '../../models/auth-token';
+import { DaffAccountRegistration } from '../../models/account-registration';
+import { DaffCustomerRegistration } from '../../models/customer-registration';
+
+/**
+ * Util pipe to get the data from an apollo result
+ */
+const unwrapResult = pipe(
+  map(({data}) =>
+    data
+  )
+);
+
+@Injectable({
+  providedIn: 'root'
+})
+export class DaffMagentoLoginService implements DaffLoginServiceInterface<DaffLoginInfo, DaffAuthToken> {
+  constructor(
+    private apollo: Apollo,
+    @Inject(DaffAuthQueryManager) public queryManager: DaffAuthQueryManagerInterface<
+      DaffAccountRegistration<DaffCustomerRegistration>,
+      DaffCustomerRegistration,
+      DaffLoginInfo
+    >,
+    @Inject(DaffAuthTransformer) public transformer: DaffAuthTransformerInterface<DaffAuthToken>
+  ) {}
+
+  login(loginInfo: DaffLoginInfo): Observable<DaffAuthToken> {
+    return this.apollo.mutate<GenerateTokenResponse>(
+      this.queryManager.generateATokenMutation(loginInfo)
+    ).pipe(
+      unwrapResult,
+      map(data => this.transformer.transform(data))
+    )
+  }
+}
```

Comment:
I think this should be generic. It is sort of halfway to being generic with the `@Inject(DaffAuthTransformer) public transformer: DaffAuthTransformerInterface<DaffAuthToken>`. But there's not really a point in injecting the transformer service with injector tokens if this whole service isn't generic. If the app dev wants to extend the `DaffLoginInfo` model, then they will need to override the transformer AND this service. If this is generic, they'll only need to override the transformer. So like:
```
export class DaffMagentoLoginService implements DaffLoginServiceInterface<
   TLoginInfo extends DaffLoginInfo, 
   TAuthToken extends DaffAuthToken,
   TCustomerRegistration extends DaffCustomerRegistration,
   TAccountRegistration extends DaffAccountRegistration
> {
  constructor(
    private apollo: Apollo,
    @Inject(DaffAuthQueryManager) public queryManager: DaffAuthQueryManagerInterface<
      TAccountRegistration<TCustomerRegistration>,
      TCustomerRegistration,
      TLoginInfo
    >,
    @Inject(DaffAuthTransformer) public transformer: DaffAuthTransformerInterface<TAuthToken>
  ) {}

  login(loginInfo: DaffLoginInfo): Observable<TAuthToken> {
    return this.apollo.mutate<GenerateTokenResponse>(
      this.queryManager.generateATokenMutation(loginInfo)
    ).pipe(
      unwrapResult,
      map(data => this.transformer.transform(data))
    )
  }
}
```
Not sure if I got everything right in that, but something like that

Reply:
Very good point. Making those changes now.